File: 2640-maximum-number-of-integers-to-choose-from-a-range-i/2640-maximum-number-of-integers-to-choose-from-a-range-i.swift
class Solution {
    func maxCount(_ banned: [Int], _ n: Int, _ maxSum: Int) -> Int {
        // Create a set to store banned numbers for quick lookup
        let bannedSet = Set(banned)
        var count = 0 // Initialize count to track the number of valid numbers included
        var currentSum = maxSum // Track the remaining sum

        // Iterate through numbers from 1 to n
        for i in 1...n {
            // Skip the number if it's in the banned set
            if bannedSet.contains(i) {
                continue
            }

            // If including the current number exceeds maxSum, return the count
            if currentSum - i < 0 {
                return count
            }

            // Deduct the current number from currentSum
            currentSum -= i

            // Increment the count
            count += 1
        }

        // Return the total count of valid numbers included
        return count
    }
}

File: 2640-maximum-number-of-integers-to-choose-from-a-range-i/2640-maximum-number-of-integers-to-choose-from-a-range-i.go
func maxCount(banned []int, n int, maxSum int) int {
     // Create a set to store banned numbers for quick lookup
    bannedSet := make(map[int]bool)
    for _, num := range banned {
        bannedSet[num] = true
    }

    count := 0 // Initialize count to track the number of valid numbers included

    // Iterate through numbers from 1 to n
    for i := 1; i <= n; i++ {
        // Skip the number if it's in the banned set
        if bannedSet[i] {
            continue
        }

        // If including the current number exceeds maxSum, return the count
        if maxSum-i < 0 {
            return count
        }

        // Deduct the current number from maxSum
        maxSum -= i

        // Increment the count
        count++
    }

    // Return the total count of valid numbers included
    return count
}